internal/impl/qdrant/integration_test.go

```diff
@@ -0,0 +1,168 @@
+// limitations under the License.
+
+package qdrant
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"testing"
+
+	pb "github.com/qdrant/go-client/qdrant"
+	"github.com/redpanda-data/benthos/v4/public/service/integration"
+	"github.com/testcontainers/testcontainers-go/modules/qdrant"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const (
+	collectionName = `redpanda`
+	template       = `
+output:
+  label: 'qdrant'
+  qdrant:
+    grpc_host: 'localhost:$PORT'
+    tls: {enabled: false}
+    id: 'root.id = $POINT_ID'
+    collection_name: $COLLECTION_NAME
+    vector_mapping: 'root = $VECTOR'
+    payload_mapping: 'root = $PAYLOAD'
+`
+)
+
+func TestIntegrationQdrant(t *testing.T) {
+	integration.CheckSkip(t)
+
+	t.Parallel()
+
+	ctx := context.Background()
+	qdrantContainer, err := qdrant.Run(ctx, "qdrant/qdrant:v1.10.1")
+	if err != nil {
+		log.Fatalf("failed to start container: %s", err)
+	}
+
+	testCases := []struct {
+		name    string
+		pointID string
+		vector  string
+		payload string
+	}{
+		{
+			name:    "Test With default dense vector",
+			pointID: `1`,
+			vector:  `[0.352,0.532,0.532]`,
+			payload: `{ "content": this.content, "some_number": 42 }`,
+		},
+		{
+			name:    "Test With sparse vector",
+			pointID: `2`,
+			vector:  `{"some_sparse": {"indices":[23,325,532],"values":[0.352,0.532,0.532]}}`,
+			payload: `{"content":"test payload"}`,
+		},
+		{
+			name:    "Test With multi vector",
+			pointID: `3`,
+			vector:  `{"some_multi": [[0.352,0.532,0.532],[0.352,0.532,0.532]]}`,
+			payload: `{"content":"test payload"}`,
+		},
+		{
+			name:    "Test With dense and sparse vector",
+			pointID: `"465213dd-3f11-4534-8daf-9fedf203549a"`,
+			vector:  `{"some_dense": [0.352,0.532,0.532],"some_sparse": {"indices": [23,325,532],"values": [0.352,0.532,0.532]}}`,
+			payload: `{"content":"test payload"}`,
+		},
+	}
+
+	containerPort, err := qdrantContainer.MappedPort(ctx, "6334/tcp")
+	if err != nil {
+		log.Fatalf("failed to get container port: %s", err)
+	}
+
+	setupCollection(ctx, fmt.Sprintf("localhost:%v", containerPort.Port()), collectionName)
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+
+			defaultQuery := func(ctx context.Context, testID, messageID string) (string, []string, error) {
+				return fmt.Sprintf(`{"content":"%v","id":%v}`, "hello world", messageID), nil, err
+			}
+
+			suite := integration.StreamTests(
+				// Is is possible to test output only without a `getFn GetMessageFunc` arg?
+				integration.StreamTestOutputOnlySendBatch(10, defaultQuery),
+			)
+			suite.Run(
+				t, template, integration.StreamTestOptPort(containerPort.Port()),
+				integration.StreamTestOptVarSet("POINT_ID", tc.pointID),
+				integration.StreamTestOptVarSet("COLLECTION_NAME", collectionName),
+				integration.StreamTestOptVarSet("VECTOR", tc.vector),
+				integration.StreamTestOptVarSet("PAYLOAD", tc.payload),
+			)
```

Comment:
This is fine, but it would be good to query qdrant after to ensure the data is written properly. Maybe it's also simpler to directly construct a writer via `newOutputWriter` then call `WriteBatch` explicitly? An example to take inspiration from is elasticsearch's integration test: https://github.com/redpanda-data/connect/blob/main/internal/impl/elasticsearch/writer_integration_test.go